### PR TITLE
feat(execution): Add ommers reward

### PR DESCRIPTION
**Pull request by rakita**
Adding ommers reward that was previously missing.

### CODE REVIEW

File: crates/executor/src/executor.rs [hunk 1]
```diff
@@ -409,24 +404,43 @@ pub fn execute<DB: StateProvider>(
...
+        let mut reward_beneficiaries: BTreeMap<H160, u128> = BTreeMap::new();
+        // Calculate Uncle reward
+        for ommer in ommers {
+            let ommer_reward = ((8 + ommer.number - header.number) as u128 * reward) >> 3;
+            *reward_beneficiaries.entry(ommer.beneficiary).or_default() += ommer_reward;
         }
-    });
+        // insert main block reward
+        *reward_beneficiaries.entry(header.beneficiary).or_default() +=
+            reward + (reward >> 5) * ommers.len() as u128;
```
Comment (rkrasiuk) [inline crates/executor/src/executor.rs:416]: can you add a link to the spec pls

**Review by rkrasiuk — APPROVED** [review]
lgtm!

File: crates/executor/src/executor.rs [hunk 2]
```diff
@@ -409,24 +404,43 @@ pub fn execute<DB: StateProvider>(
...
+                        let old = to_reth_acc(&acc);
+                        let mut new = old;
+                        new.balance += U256::from(reward);
+                        AccountInfoChangeSet::Changed { new, old }
+                    })
+                    // if account is not present append `Created` changeset
+                    .unwrap_or(AccountInfoChangeSet::Created {
+                        new: Account { nonce: 0, balance: reward.into(), bytecode_hash: None },
+                    });
+                Ok((beneficiary, changeset))
```
Comment (gakonst) [inline crates/executor/src/executor.rs:439]: this should also be part of the fn

**Review by gakonst — CHANGES_REQUESTED** [review]
Did this result in more state tests passing?

File: crates/executor/src/executor.rs (hunk 1 above)
Comment (rakita) [reply crates/executor/src/executor.rs:442]: Added the passage from yellowpaper, and OpenEthereum link to code that does this.

[the remark this answers is not in the document]
Comment (rakita) [reply crates/executor/src/executor.rs]: Moved block rewards to separate function

**Review by gakonst — APPROVED** [review]

File: crates/executor/src/executor.rs (hunk 2 above)
Comment (rakita) [reply crates/executor/src/executor.rs:462]: added it to function

**Comment by rakita** [comment]
There were no chain tests related to ommers, I just remembered that this is needed when working on something else.

**Pull request merged**